pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -26,9 +26,9 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/sets"
 	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	api "k8s.io/kubernetes/pkg/apis/core"
```

Comment:
scheduler shouldn't depend in this package (See #84879). However, I can see other binaries using this.

Maybe we can use https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/apiserver/pkg/registry/generic/matcher.go#L25

Any suggestions @liggitt ?

Reply:
I would construct the fieldset in place as `"metadata.name": node.Name`

Reply:
done.